packages/flame/lib/src/components/clip_component.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:ui';
+
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+
+/// {@template clip_component}
+/// A component that will clip its content.
+/// {@endtemplate}
+abstract class ClipComponent extends PositionComponent {
+  /// {@macro clip_component}
+  ///
+  /// Clips the canvas based its shape and size.
+  ClipComponent({
+    super.position,
+    super.size,
+    super.scale,
+    super.angle,
+    super.anchor,
+    super.children,
+    super.priority,
+  });
```

Comment:
I guess an alternative approach here could be to add the `shapeBuilder` function as a parameter (`Shape Function()`), and then declare 3 factory constructors `ClipComponent.circle`, `ClipComponent.rectangle` and `ClipComponent.polygon` instead of separate components for every shape.

PROs: (1) easier to create clip components with custom shape - no need to declare a new class, (2) easier to document a single component than several components, (3) more flexibility to change the shape at runtime.

Reply:
Yep, this is what I will do following the discussion I am having with @alestiago on a thread above.

Reply:
Oh, sorry I haven't noticed that thread.